app_test.go

```diff
@@ -105,6 +105,55 @@ func TestNewApp(t *testing.T) {
 		require.NoError(t, app.Err())
 		assert.Contains(t, g, `"fx.DotGraph" [label=<fx.DotGraph>];`)
 	})
+
+	t.Run("ProvidesWithAnnotate", func(t *testing.T) {
+		type A struct{}
+
+		type B struct {
+			In
+
+			Foo A `name:"foo"`
+			Bar A `name:"bar"`
+		}
+		app := fxtest.New(t,
+			Provide(
+				Annotated{
+					Target: func() A { return A{} },
+					Name:   "foo",
+				},
+				Annotated{
+					Target: func() A { return A{} },
+					Name:   "bar",
+				},
+			),
+			Invoke(
+				func(b B) {
+					assert.NotNil(t, b.Foo)
+					assert.NotNil(t, b.Bar)
+				},
+			),
+		)
+
+		defer app.RequireStart().RequireStop()
+		require.NoError(t, app.Err())
+	})
```

Comment:
We should probably test the following two cases as well.

- Setting Group
- Leaving both, Group and Name unset

Reply:
yeah, good point